UnityProject/Assets/Scripts/Tilemaps/Behaviours/Layers/Matrix.cs

```diff
@@ -335,13 +335,12 @@ public IEnumerable<T> Get<T>(Vector3Int localPosition, bool isServer)
 		}
 
 		var filtered = new List<T>();
-		foreach (RegisterTile t in (isServer ? ServerObjects : ClientObjects).Get(localPosition))
+		var list = (isServer ? ServerObjects : ClientObjects).Get(localPosition);
+		list.Reverse();
```

Comment:
why reverse?

Reply:
if tile gets removed or changed while checking this; it will throw an NRE.

Reply:
Couldn't you just do a for --i instead?

Reply:
both do the same thing really

Reply:
I think it's slower with the list approach, and this is a commonly called function

Reply:
Reverted after some testing. Can confirm that it is a little bit slower, nothing major, but it can add up quickly.